### PR TITLE
fix(code-frame): do not pad gutter of empty lines

### PR DESCRIPTION
**Pull request by SimenB**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Dunno 😅 It does most of https://github.com/facebook/jest/pull/10982
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  | Possibly, depending on whether the whitespace of the code frame is considered part of the contract (e.g. it'll break people snapshotting errors from Jest, but that is _not_ a supported use case on Jest's side, so it doesn't matter)
| Minor: New Feature?      | No
| Tests Added + Pass?      | Yes
| Documentation PR Link    | N/A
| Any Dependency Changes?  | No
| License                  | MIT

I don't think the trailing space adds anything.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/12567"></a>

### CODE REVIEW

File: packages/babel-code-frame/test/index.js [hunk 1]
```diff
@@ -105,8 +105,8 @@ describe("@babel/code-frame", function () {
   test("opts.highlightCode with multiple columns and lines", function () {
     // prettier-ignore
     const rawLines = [
-      "function a(b, c) {",
```
Comment (SimenB) [inline packages/babel-code-frame/test/index.js:108]: oops

Comment (SimenB) [reply packages/babel-code-frame/test/index.js:108]: ergh, shouldn't matter, I thought for a moment the trailing spaces were inside the string. So I'll leave it as is for now unless you want me to roll back

Comment (nicolo-ribaudo) [reply packages/babel-code-frame/test/index.js:108]: I'm surprised that it wasn't already fixed by prettier :thinking:

**Review by existentialism — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]
Thanks!

**Comment by nicolo-ribaudo** [comment]
@SimenB  I pushed a workaround since the Jest e2e test is failing (as expected).
We'll remove it once we publish a new version of `@babel/code-frame` and you update it in the Jest repo.

File: packages/babel-code-frame/test/index.js (hunk 1 above)
Comment (SimenB) [reply packages/babel-code-frame/test/index.js:108]: the ignore above it I assume 🙂

**Comment by SimenB** [comment]
@nicolo-ribaudo if you run against jest@master I can just apply a patch to the local install of `@babel/code-frame` until you release

**Comment by nicolo-ribaudo** [comment]
I don't want to force it on you since it's just caused by our tests, I just have to tune the regexp to make it pass :sweat_smile:

**Comment by fisker** [comment]
I'm late for this, when we replace the `EOL`

Before

```text
  1 | <LF>
  2 | <LF>
> 3 | class    a {<LF>
    |         ^^^^<LF>
> 4 |   b(  <|> ) {}<LF>
    | ^^^^^^^^^^^^^^<LF>
  5 | }<LF>
  6 | <LF>
  7 | let    x
```

After

```text
  1 |<LF>
  2 |<LF>
> 3 | class    a {<LF>
    |         ^^^^<LF>
> 4 |   b(  <|> ) {}<LF>
    | ^^^^^^^^^^^^^^<LF>
  5 | }<LF>
  6 |<LF>
  7 | let    x
```

The first one looks more "real" 😄 

Diff on Prettier PR, https://github.com/prettier/prettier/pull/10226/files#diff-802d6e48b5b7304cbc7abcb6aa1690a5419eecec01b4d971926a8ebd94548a04R85

Not important, but just want let you know.

**Comment by SimenB** [comment]
ah it's released, nice 👍 

PR updating in Jest if you wanna remove the hack made to the e2e tests here: https://github.com/facebook/jest/pull/10982

**Pull request merged**